src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,230 @@
+
+		protected override void AnalyzeSymbol(SymbolAnalysisContext context, PXContext pxContext)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			if (context.Symbol is IMethodSymbol methodSymbol)
+			{
+				if (!methodSymbol.IsStatic && methodSymbol.HasAttribute(pxContext.AttributeTypes.PXOverrideAttribute, false))
+				{
+					AnalyzeMethod(context, pxContext, methodSymbol);
+				}
+			}
+		}
+
+		private void AnalyzeMethod(SymbolAnalysisContext context, PXContext pxContext, IMethodSymbol methodSymbol)
+		{
+			// Here we know that the method is not static and has the correct attribute.
+
+			var extensionType = GetPXGraphExtension(pxContext.PXGraphExtensionTypes, methodSymbol.ContainingType);
+			var graphType = FindFirstTypeArgumentOfPxGraphExtension(extensionType);
+
+			if (graphType != null)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var allTypes = GetAllBaseTypes(pxContext.PXGraphExtensionTypes, extensionType, graphType);
+
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var hasMatchingMethod = allTypes
+					.SelectMany(t => t.GetMembers().OfType<IMethodSymbol>())
+					.Any(m => IsCompatibleMethod(m, methodSymbol));
+
+				if (!hasMatchingMethod)
+				{
+					var diagnostic = Diagnostic.Create(
+						Descriptors.PX1096_PXOverrideMustMatchSignature,
+						methodSymbol.Locations.First());
+
+					context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+				}
+			}
+			else
+			{
+				// PXOverride can only be used in a PXGraphExtension child type.
+				var diagnostic = Diagnostic.Create(
+					Descriptors.PX1096_PXOverrideContainerIsNotPXGraphExtension,
+					methodSymbol.Locations.First());
+
+				context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+			}
+		}
+
+		private static HashSet<INamedTypeSymbol> GetAllBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
+			params INamedTypeSymbol[] types)
+		{
+			var allTypes = new HashSet<INamedTypeSymbol>(SymbolEqualityComparer.Instance);
+
+			foreach (var type in types)
+			{
+				GetBaseTypes(pxGraphExtensionTypes, type, allTypes);
+			}
+
+			return allTypes;
+		}
+		private static void GetBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
+			INamedTypeSymbol typeSymbol,
+			HashSet<INamedTypeSymbol> hashSet)
+		{
+			if (typeSymbol == null || typeSymbol.SpecialType != SpecialType.None || hashSet.Contains(typeSymbol))
+			{
+				return;
+			}
+
+			hashSet.Add(typeSymbol);
+
+			var extensionType = GetPXGraphExtension(pxGraphExtensionTypes, typeSymbol.BaseType);
+			var graphType = FindFirstTypeArgumentOfPxGraphExtension(extensionType);
+
+			if (graphType != null)
+			{
+				GetBaseTypes(pxGraphExtensionTypes, graphType, hashSet);
+			}
+
+			GetBaseTypes(pxGraphExtensionTypes, typeSymbol.BaseType, hashSet);
+		}
+		private static INamedTypeSymbol FindFirstTypeArgumentOfPxGraphExtension(INamedTypeSymbol type)
```

Comment:
We have a number of extension helpers that already implement these utils in the
`Acuminator.Utilities.Roslyn.Semantic.PXGraph`. This code should be reworked by using those utils and these helpers should be removed

Reply:
This is removed as requested. Thank you.